### PR TITLE
TradeHistoryDatabase Implementation

### PR DESCRIPTION
**Pull request by Nchan8120**
has the TradeHistoryDatabase class, TradeInterface class, and updates for trade, mainUI, dummyA strategy.

### CODE REVIEW

File: src/main/java/cryptoTrader/gui/MainUI.java [hunk 1]
```diff
@@ -129,7 +130,8 @@ private MainUI() {
 			strategyNames.add(name);
 		}
 
-
+		TradeHistoryDB tradeHistroyDB = TradeHistoryDB.getInstance();
```
Comment (csweet22) [inline src/main/java/cryptoTrader/gui/MainUI.java:133]: Just a small spelling error in the variable name `tradeHistroyDB` should be `tradeHistoryDB`

File: src/main/java/cryptoTrader/utils/DummyA.java [hunk 2]
```diff
@@ -1,11 +1,22 @@
...
-        System.out.println("This strategy does nothing");
+    	//buy ETH if its more than 3,000
+    	DataFetcher fetcher = new DataFetcher();
+    	String date = java.time.LocalDate.now().toString();
+    	float price = (float) fetcher.getPriceForCoin("ETH", date);
+    	if(price > 3000) {
+    		TradeInterface tInterface = new TradeInterface();
+    		tInterface.createTrade(broker, name, "ETH", "buy", 400, price, date);
+    	}
         return null;
```
Comment (csweet22) [inline src/main/java/cryptoTrader/utils/DummyA.java:20]: Just be careful about the return value here. Function signature says it returns a Trade object but the final line says it's returning null.

File: src/main/java/cryptoTrader/utils/Trade.java [hunk 3]
```diff
@@ -1,5 +1,34 @@
 package cryptoTrader.utils;
 
 public class Trade {
-    
-}
+	String clientName;
+	String strategyName;
+	String coinTraded;
+	String action;
```
Comment (csweet22) [inline src/main/java/cryptoTrader/utils/Trade.java:7]: These probably default to being private, but can we just include private in front of all the variables just for formatting reasons? These are all private right?

File: src/main/java/cryptoTrader/utils/TradeHistoryDB.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+public class TradeHistoryDB {
+
+	private ArrayList<Object> db = new ArrayList<Object>();
+	private HashMap<String,Object[]> numTrades = new HashMap<String, Object[]>();//index 0 is the number of trades, index 1 is the strategy
+	
+	private static TradeHistoryDB database = null;
+	
+	public TradeHistoryDB() {
+	
+	}
```
Comment (csweet22) [inline src/main/java/cryptoTrader/utils/TradeHistoryDB.java:15]: Nitpicking (You can choose to ignore or change this): Can we format this just as `public TradeHistoryDB() {}` if it is going to be empty?

**Comment by csweet22** [comment]
What code do you need before we can implement the TradeHistory within the MainUI?

i.e. What do you need to get it displaying the history correctly?

**Review by csweet22 — CHANGES_REQUESTED** [review]

**Comment by Nchan8120** [comment]
> What code do you need before we can implement the TradeHistory within the MainUI?
> 
> i.e. What do you need to get it displaying the history correctly?

do you mean the table and histogram? if so that is Nick's part of the code. I'm focused on the process of making a trade

**Review by csweet22 — APPROVED** [review]

**Pull request merged**